homeassistant/helpers/condition.py

```diff
@@ -623,12 +620,12 @@ def sun(
         condition_trace_update_result(wanted_time_after=wanted_time_after)
         return utcnow < wanted_time_before or utcnow > wanted_time_after
 
-    if sunrise is None and SUN_EVENT_SUNRISE in (before, after):
+    if sunrise is None and has_sunrise_condition:
```

Comment:
See now with this change it seems a bit more obvious that there are some edge case problems here:

Super rare problem, we're in the polar regions on a day with no sunrise, but only a sunset and we have the following config:

```
after: sunrise
before: sunset
```

I would expect the condition to be true, for the entire day until sunset, but this statement just says "No sunrise, so false".  

It may be worth ignoring the no sunrise sunset cases and just making sure everything else works as expected and then either say "this condition doesn't handle no sunrise/set, use above/below horizon instead" or figure out how to get them back in.  The offsets are the thing that makes using above/below horizon harder.